x264/patches/input.patch

```diff
@@ -0,0 +1,47 @@
+
+--- x264_org/input/input.c	2024-09-05 11:57:22.943594154 +0200
++++ x264/input/input.c	2024-09-05 11:57:41.344630765 +0200
+@@ -227,6 +227,15 @@
+     }
+ #else
+     size_t padded_size = size + MMAP_PADDING;
++    /**
++     * PHOENIX-RTOS PATCH
++     * 
++     * Phoenix implementation of mmap() fails if `size` is not aligned to page size
++     * It is important to remove this patch when this issue is resolved.
++     * 
++     * https://github.com/phoenix-rtos/phoenix-rtos-project/issues/1155
++     */
++    padded_size = (1 + padded_size / SIZE_PAGE) * SIZE_PAGE;
```

Comment:
if this is a hot path - use bit operations instead of division (`padded_size = (padded_size + SIZE_PAGE - 1) & ~SIZE_PAGE`. 
Applies below also.

Reply:
Thanks! Used the following math:
`padded_size = (padded_size + (SIZE_PAGE - 1)) & ~(SIZE_PAGE - 1)`

Reply:
yeah, sorry, missed the last `-1`